### DISCUSSION TITLE
:cached or :delegated?

### DISCUSSION THREAD

**Discussion by r-martins** [markshust/docker-magento#546] [Q&A] [answered]
If [this answer](https://stackoverflow.com/a/63437557/529403) is correct, shouldn't lines like this one be `:delegated` instead of `:cached`?
https://github.com/markshust/docker-magento/blob/1d9b25cb209b4faab63a2b380108b83f7e6d8b46/compose/docker-compose.dev.yml#L21

Quoting:

> TLDR:
> 
>     Use cached: when the host performs changes, the container is in read only mode.
>     Use delegated: when docker container performs changes, host is in read only mode.
>     Use default: When both container and host actively and continuously perform changes on data.

PS: this is really a dumb question of someone who just gave a first look into the subject. Please delete and ignore if makes no sense. No heart feelings. ;)

**Comment by markshust** [comment, marked as answer]
It's not a dumb question at all. The `:delegated` and `:cached` flags are actually the same. I changed this a while back per Docker's recommendations that the `:delegated` flag will be reserved for future use, and I didn't want any of their updates to break this project. They don't document this anywhere, and still have outdated info as the above. Rest assured, the `:cached` flag has the highest speed and is very stable 😁

**Reply by r-martins** [reply]
Thanks, @markshust. I couldn't expect less from you. :)
And also thanks for your good work with this project.
